openedx/core/djangoapps/content_libraries/docs/decisions/0003-course-content-independence.rst

```diff
@@ -0,0 +1,48 @@
+3. Course Content should not depend on Library Content
+######################################################
+
+Status
+******
+
+Accepted
+
+
+Context
+*******
+
+When a course uses content from a library via a ``LibraryContentBlock``, it creates a course-local copy of the library's blocks. For each library block, it will also store certain metadata in the ModuleStore that connects the course copy with the original in the library, such as the usage key and version of the original, as well as the library-suggested values for various fields (e.g. ``num_attempts=10``).
```

Comment:
This describes the situation with modulestore. Would it be any different with Learning Core? (I had hoped that we'd be able to do less copying and more referencing, for example.)

Reply:
I'm not sure. I initially hated the fact that we were copying so much instead of referencing, but I've come to appreciate the decoupling this actually gave when considering something like the v1 -> v2 upgrade process, because it would mean that old content that was never going to be used again didn't need to be upgraded. I think that the use cases we have for cross-instance import/export also point to an area where copying is the better solution.

Internally, I think it makes sense to have a schema that allows for references + copies where we cross ownership/lifecycle boundaries. But those boundaries are really fuzzy to me at the moment. I'm strongly against having a 1:1 straight mapping of v1 -> v2 libraries because I think it's a terrible missed opportunity. But if not that, then what? Manually created v2 libraries that import multiple v1 libraries? An implicit, org-level shared content library maybe? I dunno.